python-checks/src/main/java/org/sonar/python/checks/TfPyTorchSpecifyReductionAxisCheck.java

```diff
@@ -0,0 +1,122 @@
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.quickfix.TextEditUtils;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6929")
+public class TfPyTorchSpecifyReductionAxisCheck extends PythonSubscriptionCheck {
+
+  private static final Set<String> TF_REDUCTION_FUNCTIONS = new HashSet<>(Arrays.asList("reduce_all", "reduce_mean", "reduce_any",
+    "reduce_euclidean_norm", "reduce_logsumexp",
+    "reduce_max", "reduce_min", "reduce_prod", "reduce_std", "reduce_sum", "reduce_variance"));
+  private static final Set<String> TF_REDUCTION_FUNCTIONS_FQN = new HashSet<>();
+  private static final String TF_MESSAGE = "Provide a value for the axis argument.";
+  public static final String AXIS_PARAMETER = "axis";
+  public static final int AXIS_PARAMETER_POSITION = 1;
+
+  static {
+    for (String reductionFunction : TF_REDUCTION_FUNCTIONS) {
+      TF_REDUCTION_FUNCTIONS_FQN.add("tensorflow.math." + reductionFunction);
+      TF_REDUCTION_FUNCTIONS_FQN.add("tensorflow.tf." + reductionFunction);
+    }
+  }
+
+  private static final String PY_TORCH_MESSAGE = "Provide a value for the dim argument.";
+  public static final String DIM_PARAMETER = "dim";
+  public static final int NO_POSITIONAL_ARG = -1;
+
+  /**
+   * Contains a list of reduction functions with a {@code dim} parameter and the position of the dim argument in that function.
+   */
+  private static final Map<String, Integer> PY_TORCH_REDUCTION_FUNCTIONS_DIM_POS = Map.ofEntries(
+    Map.entry("torch.argmin", 1),
+    Map.entry("torch.aminmax", NO_POSITIONAL_ARG),
+    Map.entry("torch.nanmean", 1),
+    Map.entry("torch.mode", 1),
+    Map.entry("torch.norm", 2),
+    Map.entry("torch.quantile", 2),
+    Map.entry("torch.nanquantile", 2),
+    Map.entry("torch.std", 1),
+    Map.entry("torch.std_mean", 1),
+    Map.entry("torch.unique", 4),
+    Map.entry("torch.unique_consecutive", 3),
+    Map.entry("torch.var", 1),
+    Map.entry("torch.var_mean", 1),
+    Map.entry("torch.count_nonzero", 1)
+  );
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, TfPyTorchSpecifyReductionAxisCheck::checkCallExpr);
+  }
+
+  private static void checkCallExpr(SubscriptionContext context) {
+    CallExpression callExpression = (CallExpression) context.syntaxNode();
+    Symbol symbol = callExpression.calleeSymbol();
+    if (symbol != null && !hasUnpackArgument(callExpression.arguments())) {
+      if (isTfReductionMissingAxisArg(symbol, callExpression)) {
+        PreciseIssue issue = context.addIssue(callExpression.callee(), TF_MESSAGE);
+        createTfQuickFix(callExpression).ifPresent(issue::addQuickFix);
+      }
+
+      if (isPyTorchReductionMissingDimArg(symbol, callExpression)) {
+        context.addIssue(callExpression.callee(), PY_TORCH_MESSAGE);
+      }
+    }
+  }
+
+  private static boolean hasUnpackArgument(List<Argument> arguments) {
```

Comment:
I was not sure but did you already migrated this method to TreeUtils. I think I saw this one several times now.